### PR TITLE
Soe soeop 368

### PR DESCRIPTION
**Pull request by mdyoung3**
# READY FOR REVIEW
Ready for review.

# Summary
- Docksal configuration scripts that set up local docker containers and build the site. 

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- 3/10
- Affects SOE local setup.

# Review Tasks

## Setup tasks and/or behavior to test

1. Clone site locally
2. `Cd` into project root folder
3. Read the README.md file located in .`docroot` folder for steps to set up site locally.

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Associated Issues and/or People
- https://stanfordits.atlassian.net/browse/SOEOP-368

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

**Review by jbickar — CHANGES_REQUESTED** [review]
This PR should be submitted to the `soe-develop-mc` branch.

**Comment by mdyoung3** [comment]
@jbickar Switched base branch for PR.

File: .docksal/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+Running `grunt settings` results in an error that the same files are already in the folder. Skipping step.
+
+##Links
+https://github.com/SU-SOE/grunt-drupal-tasks/blob/SOE-develop-mc/docs/soe_build_notes.md
+https://stackoverflow.com/questions/47008159/how-to-downgrade-node-version
+https://www.npmjs.com/package/n
+https://phase2.github.io/grunt-drupal-tasks/10_BUILD/
+https://www.drupal.org/node/1401522
+
+
```
Comment (jbickar) [inline .docksal/README.md]: I believe we discussed dropping these, as they are internal notes.

File: .docksal/src_files/Acquia_settings/.acquia/cloudapi.conf [hunk 2]
```diff
@@ -0,0 +1 @@
+{"email":"mdyoung3@stanford.edu","key":"gOoBtv57btCoLt0EXa8yzmas\/ESORvy2dV1LVQ10W+V7TdCyVQuUR28GgvMeQV9bH5zFOuKqsjP4"}
```
Comment (jbickar) [inline .docksal/src_files/Acquia_settings/.acquia/cloudapi.conf]: Probably don't want to commit an API key to a public repo. And you now probably should revoke that and re-generate.

File: .docksal/src_files/Acquia_settings/.drush/cloudapi.acquia.com.pem [hunk 3]
```diff
@@ -0,0 +1,99 @@
+-----BEGIN CERTIFICATE-----
```
Comment (jbickar) [inline .docksal/src_files/Acquia_settings/.drush/cloudapi.acquia.com.pem]: Why do we need the certificate?

File: .docksal/src_files/Acquia_settings/.drush/hsmrc.aliases.drushrc.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
+<?php
```
Comment (jbickar) [inline .docksal/src_files/Acquia_settings/.drush/hsmrc.aliases.drushrc.php]: I would only commit the `swssoe` drush aliases for Acquia.

File: .docksal/src_files/Acquia_settings/.drush/stanfordfsh.aliases.drushrc.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
+<?php
```
Comment (jbickar) [inline .docksal/src_files/Acquia_settings/.drush/stanfordfsh.aliases.drushrc.php]: I would only commit the `swssoe` drush aliases for Acquia.

File: .docksal/src_files/Acquia_settings/.drush/swssoe.aliases.drushrc.php [hunk 6]
```diff
@@ -0,0 +1,122 @@
+<?php
```
Comment (jbickar) [inline .docksal/src_files/Acquia_settings/.drush/swssoe.aliases.drushrc.php:1]: Yes, these. I won't leave comments on each of the other alias files. 😄

File: .docksal/src_files/Acquia_settings/README-acquiacloud.txt [hunk 7]
```diff
@@ -0,0 +1,3 @@
+This folder contains two hidden folders named .acquia and .drush. Both of these folders should be placed in the $HOME directory on your system. The .acquia folder contains your Acquia Cloud API credentials and the .drush folder contains your Drush aliases for the sites you currently have access to on Acquia Cloud.
```
Comment (jbickar) [inline .docksal/src_files/Acquia_settings/README-acquiacloud.txt]: I don't think we need this file.

File: .docksal/src_files/Gruntfile.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+module.exports = function(grunt) {
+  // Load all plugins and tasks defined by the grunt-drupal-tasks package.
+  require('grunt-drupal-tasks')(grunt);
+  grunt.loadNpmTasks('grunt-deploy-site');
+  grunt.loadNpmTasks('grunt-symlink');
+  grunt.config.set('deploy_site.development.options.branch', 'SOE-SOEOP-368');
```
Comment (jbickar) [inline .docksal/src_files/Gruntfile.js]: The default here probably should be `soe-develop-mc`. Future improvement would be to take this as an argument or from a variable.

File: .docksal/commands/dev-mode [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+#: exec_target = cli
+
+DOCROOT_PATH="${PROJECT_ROOT}/${DOCROOT}"
+
+cd "$DOCROOT_PATH" || return
+drush variable-set preprocess_js 0
+drush variable-set preprocess_css 0
+
+find "$DOCROOT_PATH/sites/default/files" -type f -exec sudo chmod 644 {} \;
+find "$DOCROOT_PATH/sites/default/files" -type d -exec sudo chmod 755 {} \;
```
Comment (jbickar) [inline .docksal/commands/dev-mode]: Please add a blank line to the end of each bash file.

**Review by jbickar — CHANGES_REQUESTED** [review]
Please review comments and make the associated changes.

File: .docksal/commands/dev-mode (hunk 9 above)
Comment (mdyoung3) [reply .docksal/commands/dev-mode]: Done

File: .docksal/README.md (hunk 1 above)
Comment (mdyoung3) [reply .docksal/README.md]: Yes, however, I'm reluctant to remove them. They contain some research regarding the setting up Grunt for this project. (E.g., that it will only work with an older version of Node.)

I'd recommend transferring relevant information therein to the Build Notes. Re: https://github.com/SU-SOE/grunt-drupal-tasks/blob/SOE-develop-mc/docs/soe_build_notes.md

File: .docksal/src_files/Acquia_settings/.acquia/cloudapi.conf (hunk 2 above)
Comment (mdyoung3) [reply .docksal/src_files/Acquia_settings/.acquia/cloudapi.conf]: Agreed. I'll remove it. 

Probably have to update the README to include instructions for setting up AC tokens.

File: .docksal/src_files/Acquia_settings/.drush/swssoe.aliases.drushrc.php (hunk 6 above)
Comment (mdyoung3) [reply .docksal/src_files/Acquia_settings/.drush/swssoe.aliases.drushrc.php:1]: Sounds good.

File: .docksal/commands/dev-mode (hunk 9 above)
Comment (sherakama) [reply .docksal/commands/dev-mode]: You should be able to configure your IDE to do this automagically.

File: .docksal/commands/grunt-init [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+n 10.22.1
+npm install -g grunt-cli
+
+# Copying install files.
+rsync -az "$PROJECT_ROOT/.docksal/src_files/Gruntconfig.json" "$PROJECT_ROOT/soe"
+rsync -az "$PROJECT_ROOT/.docksal/src_files/Gruntfile.js" "$PROJECT_ROOT/soe"
+rsync -az "$PROJECT_ROOT/.docksal/src_files/settings.local.php" "$PROJECT_ROOT/soe/src/sites/default"
+rsync -az "$PROJECT_ROOT/.docksal/src_files/Acquia_settings/.drush/"* "$HOME/.drush"
+rsync -az "$PROJECT_ROOT/.docksal/src_files/soe.aliases.drushrc.php"* "$HOME/.drush"
+rsync -az -r "$PROJECT_ROOT/.docksal/src_files/Acquia_settings/.acquia" "$HOME"
```
Comment (jbickar) [inline .docksal/commands/grunt-init:33]: This line now can be removed as there's nothing in there.

File: .docksal/README.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+2. Checkout new branch, e.g., `git checkout -b SOE-jiraticket-shortdescription`
+3. Edit the following files, which are located ./docksal
+    1. Add branch name to line six in Gruntfile.js.
+    2. Edit soe.aliases.drushrc.php, making sure the slugs path to build/html is correct.
+    3. Optional: edit settings.local.php, updating the `base_url` var. Note that the `base_url` is set based on the directory name of this repository, e.g., `grunt-drupal-tasks.docksal`.
+4. If you use a key other than `id_rsa` to connect to Acquia, run `fin ssh-key add <keyname>` (e.g., `fin ssh-key add id_rsa_acquia_4096`)
+5. Run `fin init`
+
+
+Building Notes:
```
Comment (jbickar) [inline .docksal/README.md]:
```suggestion
## Building/Troubleshooting Notes:
```

Comment (jbickar) [reply .docksal/README.md]: Per discussion, move this below the `## Links` section, below.

File: .docksal/README.md [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+
+Be sure to add path to src. Refer to the following:
+
+```
+"srcPaths": {
+           "make": "src/engineering.make",
+           "drupal": "src"
+         },
+```
+
```
Comment (jbickar) [inline .docksal/README.md]: Per discussion, drop this from the README for Docksal, and submit a separate PR to add this change to the `soe/Gruntconfig.json` file

**Comment by mdyoung3** [comment]
Hi @jbickar 

I pushed up the fixes with the changes/removals/edits per your feedback.

**Comment by jbickar** [comment]
@mdyoung3, is this ready for review again? Can you resolve the `.gitignore` conflict if so?

We should get this in so that I can review #24 (I'm reviewing it, it's just that I'm working off of multiple branches).

**Comment by mdyoung3** [comment]
HI @jbickar 

Sorry I missed your message. The gitignore is merged and it is ready for review. 

Best,
Marc Young

**Review by jbickar — APPROVED** [review]
GTG
